### ISSUE TITLE
Path for .flow location should be relative when using Docker Storage through SSH

### ISSUE THREAD

**Issue by thomasfrederikhoeck** [PrefectHQ/prefect#3023]
## Description
When using Docker storage on windows with SHH to a docker daamon the build context is copied over, with regards to the placement of the Dockerfile which is located in `project` but prefect inserts the whole path for the file on the local Windows machine. However this doesn't exists in the build context.
```

Step 16/21 : RUN mkdir -p /opt/prefect/
 ---> Running in 4f18a09c30b0
Removing intermediate container 4f18a09c30b0
 ---> 4d6f5c0b81eb
Step 17/21 : COPY C:\Users\tfh\project\tmp3dwq67c0\etl.flow /opt/prefect/flows/etl.prefect
COPY failed: stat /var/lib/docker/tmp/docker-builder766472190/C:Userstfhprojecttmp3dwq67c0etl.flow: no such file or dir
ectory
Traceback (most recent call last):
  File "build_flows.py", line 31, in <module>
    build_image(all_flows)
  File "build_flows.py", line 23, in build_image
    storage = storage.build(push=push)
  File "C:\Users\tfh\Miniconda3\envs\prefect\lib\site-packages\prefect\environments\storage\docker.py", line 315, in build
    self._build_image(push=push)
  File "C:\Users\tfh\Miniconda3\envs\prefect\lib\site-packages\prefect\environments\storage\docker.py", line 38  File "C:\Users\tfh\Miniconda3\envs\prefect\lib\site-packages\prefect\environments\storage\docker.py", line 31
5, in build
    self._build_image(push=push)
  File "C:\Users\tfh\Miniconda3\envs\prefect\lib\site-packages\prefect\environments\storage\docker.py", line 38
0, in _build_image
    "Your docker image failed to build!  Your flow might have "
ValueError: Your docker image failed to build!  Your flow might have failed one of its deployment health checks
 - please ensure that all necessary files and dependencies have been included.

```

## Expected Behavior
I expect the COPY statement to be relative such that the stamtement is `COPY etl.flow /opt/prefect/flows/etl.prefect`

**Comment by pravindahal** [comment]
Also, the paths for healthcheck.py and custom files if any are affected. I will submit a PR.

**Comment by thomasfrederikhoeck** [comment]
Yes, you are right @pravindahal :-) I think you should be able to detect it simply by seeing if `self.base_url` starts with` ssh://` and add an extra check to https://github.com/PrefectHQ/prefect/blob/e83e838fc8e8ac10558758ffb9f0b19edce6abec/src/prefect/environments/storage/docker.py#L343

**Comment by joshmeek** [comment]
Closed by #3044
